### PR TITLE
Remove deprecated call to `DSL.jsonObject`

### PR DESCRIPTION
**Pull request by apmasell**
Replaces deprecated call to JSON-encoded string.

- [NA] Includes a change file
- [NA] Updates developer documentation

### CODE REVIEW

File: vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java [hunk 1]
```diff
@@ -1112,18 +1118,18 @@ private void createAnalysisRecords(
...
-          .forEach(
-              result -> {
-                try {
-                  jsonGenerator.writeRawValue(result.value1().data());
-                } catch (IOException e) {
-                  throw new RuntimeException(e);
-                }
-              });
+    context
+        .select(DSL.jsonObject(fields))
```
Comment (avarsava) [inline vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java:1122]: i thought this pr is to remove this?

Comment (apmasell) [reply vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java:1122]: Weirdly, no. Only certain calls to `DSL.jsonObject` are deprecated. The array form is. The one that takes `JSONEntry` is fine.
https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/impl/DSL.html#jsonbObject(org.jooq.Field...)

Comment (avarsava) [reply vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java:1122]: Ohh I see. the formatting made github not highlight the other change and i missed it. sounds good!

**Review by avarsava — APPROVED** [review]

File: vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java (hunk 1 above)
Comment (apmasell) [reply vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/Main.java:1122]: Ah, sorry.

**Review by grapigeau — APPROVED** [review]

**Pull request merged**